lib/statsd.js

```diff
@@ -160,7 +106,7 @@ const Client = function (host, port, prefix, suffix, globalize, cacheDns, mock,
   // will gracefully (attempt) to re-open the socket with a small delay
   // options.udsGracefulRestartRateLimit is the minimum time (ms) between creating sockets
   // does not support options.isChild (how to re-create a socket you didn't create?)
-  if (!options.isChild && options.protocol === 'uds' && options.udsGracefulErrorHandling) {
+  if (!options.isChild && options.protocol === PROTOCOL.UDS && options.udsGracefulErrorHandling) {
```

Comment:
Would be nice to abstract this piece in the uds transport, but perhaps a story for another PR. This is very special case for uds only.